Blaze: Add entry point for dashboard

momo-ozawa opened the pull request:
Part of #20090

- Adds an entry point for the dashboard menu item
- Button actions will be addressed in a different PR
- Analytics will be addressed in a different PR
- Figma: cEGtlsm6xijEswaKkrQrIj-fi-477%3A5675

## How to test

0. Make sure that Blaze is enabled via the debug menu
1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. Switch to the dashboard
3. ✅ A Blaze dashboard card is displayed
4. Switch to a site that's not eligible for Blaze
5. ✅ A Blaze dashboard card is NOT displayed

<img src="https://user-images.githubusercontent.com/6711616/220160874-a3d59666-e3cc-4b1f-b7b8-74129a4d1fdc.png" width=200>

## Regression Notes
7. Potential unintended areas of impact
n/a

8. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

9. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Add: Blaze dashboard card views

Commit:
Add: display blaze dashboard card

Commit:
Fix: hide icon image view is image is nil

Commit:
Add: flame image

hassaanelgarem reviewed (approved):
Works as described 🚀

Commit:
Merge branch 'trunk' into task/20090-dashboard-entry-point

tiagomar commented:
👋 @momo-ozawa!

I noticed an issue: the first time loading an eligible site, the card is not displayed until the Dashboard is reloaded.

| **Pull to refresh** | **Switching sites** | **Switching tabs** | 
| - | - | - |
| <video width="200" src="https://user-images.githubusercontent.com/42008628/221306789-4d3d9923-215a-4be6-8a76-4bf06c8a88c8.mov"> |<video width="200" src="https://user-images.githubusercontent.com/42008628/221307641-0d6282d8-f7cf-4ccc-98ab-1d68367b7791.mov">  | <video width="200" src="https://user-images.githubusercontent.com/42008628/221306933-1f4e8060-ed5b-410c-a787-3ecf8d2659d3.mov"> |

momo-ozawa commented:
Hey @tiagomar 👋 
I'm planning on addressing issues with fetching / updating the blaze status in a different PR. I'll be sure to tag you once I push the PR - thanks!
